services/kube-prometheus-stack/46.8.0/etcd-metrics-proxy/kube-rbac-proxy.yaml

```diff
@@ -75,25 +75,32 @@ spec:
       initContainers:
         - name: init
           image: docker.io/library/busybox:1
-          command: ["/bin/chown", "-R", "65532:65532", "/etc/certs/etcd"]
+          command: ["/bin/sh", "-c"]
+          args:
+            - >
+              cp /etc/certs/etcd/ca.crt /etc/certs/etcd/healthcheck* /etc/kube-rbac-proxy/etcd/ &&
```

Comment:
I believe this copy would need to get 'refreshed' every time the etcd certs expire, right?

We use kubeadm to install kubernetes, which creates (by default) certs with a 1 year expiry.
However, in 2.4 we added a feature that allows the customer to set a custom renewal frequency.

https://docs.d2iq.com/dkp/2.5/aws-certificate-renewal

I am not sure if we have a mechanism to restart pods when the cluster PKI gets updated - we do have the 'reloader', but I believe it can only watch secrets/certs...

Reply:
Turns out the control-plane nodes are restarted during certificate rotation, so the pods will be recreated automatically ([source](https://d2iq.slack.com/archives/C01MPKVBB5E/p1688737943790949?thread_ts=1688726068.985899&cid=C01MPKVBB5E)).